Throw if ruleId is not a string

sonicdoe opened the pull request:
This makes it more obvious where and why the error occurred.

Before:

```
TypeError: Cannot read property 'split' of null
```

After:

```
TypeError: ruleId must be a string, got object
```

sonicdoe committed:
Throw if ruleId is not a string

j-f1 commented:
Maybe add a `, got '${ruleId}'` part to the message so the developer can see what they sent?

sonicdoe commented:
I’ve changed it to ``new TypeError(`ruleId must be a string, got ${typeof ruleId}`)``.

Arcanemagus commented:
@j-f1 That would help in the specific case of the test being added (`null`), but doesn't help if an _actual_ Object is passed in, should we do `, got '${JSON.stringify(ruleId)}'`? Or just leave it off entirely?

Arcanemagus reviewed (approved):
This works just fine as is, but it _might_ be nice to include @j-f1's suggestion.

Arcanemagus commented:
Note: The specs are only failing on Node.js v0.12. Since that is no longer maintained I've put up #19 for removing it from the test matrix.

jfmengels reviewed (approved):
LGTM, thanks @sonicdoe.
I've just merged #19 so tests should pass. You might need to rebase to make them pass though.
I also agree that @j-f1's suggestion would be a nice addition.

sonicdoe commented:
Rebased.

Also, in case you missed it, I have already incorporated @j-f1’s suggestion (see https://github.com/jfmengels/eslint-rule-documentation/pull/18#issuecomment-319788364).

Arcanemagus commented:
That just shows the _type_ of the thing passed in, but that's probably good enough, and certainly better than nothing so I'm fine with merging as is, thanks!

Arcanemagus commented:
Published in v1.0.15.

sonicdoe commented:
I figured it would be more consistent with the statement in front (“must be a string”) and avoids any confusion where a variable might look like a string but isn’t.

Nevertheless, I’m fine with either version and even Node.js’s core modules are inconsistent in that regard (**edit:** see https://github.com/nodejs/node/issues/14614).
